packages/video_player/video_player/lib/video_player.dart

```diff
@@ -14,10 +14,10 @@ import 'package:video_player_platform_interface/video_player_platform_interface.
 export 'package:video_player_platform_interface/video_player_platform_interface.dart'
     show DurationRange, DataSourceType, VideoFormat;
 
-// This will clear all open videos on the platform when a full restart is
-// performed.
-// ignore: unused_element
-final VideoPlayerPlatform _ = VideoPlayerPlatform.instance..init();
+final VideoPlayerPlatform _videoPlayerPlatform = VideoPlayerPlatform.instance
+  // This will clear all open videos on the platform when a full restart is
+  // performed.
+  ..init();
```

Comment:
I don't like this hack in general, but we don't have a better alternative right now. I filed https://github.com/flutter/flutter/issues/48586 to expose an engine restart hook to the plugin's platform code.